### ISSUE TITLE
Siconos doesn't compile with libclapack-dev

### ISSUE THREAD

**Issue by radarsat1** [siconos/siconos#83]
I was recently building Siconos on a new Ubuntu 16.10 container and
accidentally installed `libclapack-dev` instead of `libatlas-dev`.
I got the following error:

~~~
$ make -C ~/build/
make: Entering directory '/home/siconos/build'
make[1]: Entering directory '/home/siconos/build'
make[2]: Entering directory '/home/siconos/build'
make[2]: Leaving directory '/home/siconos/build'
[  4%] Built target externals
make[2]: Entering directory '/home/siconos/build'
make[2]: Leaving directory '/home/siconos/build'
make[2]: Entering directory '/home/siconos/build'
[  5%] Building C object numerics/CMakeFiles/numerics.dir/src/NonSmoothNewton.c.o
In file included from /home/siconos/siconos/externals/blas_lapack/SiconosAtlasLapack.h:30:0,
                 from /home/siconos/siconos/externals/blas_lapack/SiconosLapack.h:76,
                 from /home/siconos/siconos/numerics/src/NonSmoothNewton.c:20:
/usr/include/clapack.h:6:29: error: unknown type name 'integer'
/* Subroutine */ int caxpy_(integer *n, complex *ca, complex *cx, integer *
                            ^~~~~~~
/usr/include/clapack.h:6:41: error: unknown type name 'complex'
/* Subroutine */ int caxpy_(integer *n, complex *ca, complex *cx, integer *
                                        ^~~~~~~

...
~~~

(about a million of these..)

Do we want Siconos to work using `libclapack-dev`?  In any case,
installing `libatlas-dev` fixed it.

**Comment by radarsat1** [comment]
By "fixed it", I meant cmake gave no more errors, but there were still `make` errors which I mention in issue #84.

After resolving those and finally getting Siconos compiled and installed (i.e. successful `make install`), I got the following error running a mechanics-io Python script:

~~~
Traceback (most recent call last):
  File "chute_con_rocas.py", line 3, in <module>
    from siconos.io.mechanics_io import Hdf5
  File "/home/siconos/.local/lib/python2.7/site-packages/siconos/io/mechanics_io.py", line 19, in <module>
    from siconos.mechanics.collision.tools import Contactor
  File "/home/siconos/.local/lib/python2.7/site-packages/siconos/mechanics/collision/__init__.py", line 2, in <module>
    from base import *
  File "/home/siconos/.local/lib/python2.7/site-packages/siconos/mechanics/collision/base.py", line 28, in <module>
    _base = swig_import_helper()
  File "/home/siconos/.local/lib/python2.7/site-packages/siconos/mechanics/collision/base.py", line 24, in swig_import_helper
    _mod = imp.load_module('_base', fp, pathname, description)
ImportError: /home/siconos/.local/lib/libsiconos_numerics.so.4.1.0: undefined symbol: clapack_dpotrf
~~~

I tried apt-get removing libclapack-dev and erasing the build directory and running CMake from scratch.  Still get this error.

**Comment by radarsat1** [comment]
Installing `libatlas-base-dev`, seems to have fixed the problem.

**Comment by vacary** [comment]
in siconos/Build/Pipol/rc.ubuntu, there is 

`if [ "$LESS_THAN_1404" -eq 0 ]; then
# 14.04 onwards
apt-get -y install cmake gfortran gcc doxygen libboost-graph-dev libboost-filesystem-dev libboost-serialization-dev libopenblas-dev liblapacke-dev libgmp3-dev libcppunit-dev python-pytest python-numpy python-scipy python-dev  make g++ git gccxml python-pygccxml python-lxml swig2.0`

Is it still working or not ? If not, we have to update the recipe ? By the way, the main problem in installation  Ubuntu/debian is related to the packing of BLAS/LAPACK that is really weird. Should we embed BLAS?LAPACK ?

**Comment by radarsat1** [comment]
I'll try again with that exact formula tomorrow.  My overall point though was more that we should get CMake to catch these problems, because during the make stage it gets extremely confusing..

**Comment by fperignon** [comment]
I commit a fix in FindLAPACK.cmake. The wrong header (clapack.h instead of lapacke.h) was searched. Hope it works now. If not you can try -DWITH_LAPACK='lapacke'.

By the way the rc.ubuntu file is probably out of date and not used anymore.
The "real" recipes are those in CI/config/siconos.yml. Can you confirm that Maurice?

**Comment by radarsat1** [comment]
Thank you Frank! Confirmed fixed in 39ba52de603789dd97858ef2dc48b86419883147.  The suggested WITH_LAPACK option also worked.
I would propose also some kind of try-compile check to ensure compiling against LAPACK works as expected, but let's consider it fixed.
